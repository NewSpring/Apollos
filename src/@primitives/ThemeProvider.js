import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_THEME, THEME_PROPS } from './constants';


export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.shape(THEME_PROPS),
  };

  static defaultProps = {
    children: null,
    theme: DEFAULT_THEME,
  };

  static childContextTypes = {
    theme: PropTypes.shape(THEME_PROPS),
  };

  getChildContext = () => ({
    theme: Object.assign({}, DEFAULT_THEME, this.props.theme),
  })

  render() {
    return Children.only(this.props.children);
  }
}
