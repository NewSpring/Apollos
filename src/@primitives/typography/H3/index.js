import React from 'react';
import { Text } from 'react-native';
import { compose, pure, setPropTypes } from 'recompose';
import PropTypes from 'prop-types';
import styled from '@primitives/styled';
import rem from '@utils/remUnit';
import verticalRhythm from '@utils/verticalRhythm';

const enhance = compose(
  pure,
  setPropTypes({
    children: PropTypes.node,
    style: Text.propTypes.style,
  }),
);

const StyledH3 = styled(({ theme }) => ({
  fontSize: rem(1.8),
  lineHeight: verticalRhythm(1.8, 1.145),
  fontWeight: '700',
  color: theme.baseFontColor,
}))(Text);

const H3 = enhance(({
  children,
  style: styleProp = {},
  ...otherProps
}) => (
  <StyledH3
    style={styleProp}
    {...otherProps}
  >
    {children}
  </StyledH3>
));

export default H3;
