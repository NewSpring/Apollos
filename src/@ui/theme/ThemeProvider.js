import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import createTheme, { THEME_PROPS } from './createTheme';


export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  static defaultProps = {
    children: null,
    themeInput: {},
  };

  static childContextTypes = {
    theme: PropTypes.shape(THEME_PROPS),
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  getChildContext = () => ({
    theme: createTheme(this.props.themeInput),
    themeInput: this.props.themeInput,
  })

  render() {
    return Children.only(this.props.children);
  }
}
