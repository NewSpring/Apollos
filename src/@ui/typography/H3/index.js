import { Platform, Text } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import styled from '@ui/styled';
import { withPlaceholder, Typography } from '@ui/Placeholder';

const styles = styled(
  ({ theme }) => ({
    fontSize: theme.helpers.rem(1.8),
    fontFamily: theme.typography.fontFamilySans.bold.default,
    color: theme.colors.text.primary,
    ...Platform.select({
      ios: {
        paddingTop: theme.helpers.rem(0.3),
        paddingBottom: theme.helpers.rem(0.15),
        lineHeight: theme.helpers.verticalRhythm(1.8, 1.14),
      },
      android: {
        lineHeight: theme.helpers.verticalRhythm(1.8, 1.34),
      },
      web: {
        paddingTop: theme.helpers.rem(0.3),
        paddingBottom: theme.helpers.rem(0.1),
        lineHeight: theme.helpers.verticalRhythm(1.8, 1.15),
        WebkitFontSmoothing: 'antialiased',
        MozOsxFontSmoothing: 'grayscale',
      },
    }),
  }),
  'H3',
);

const H3 = compose(
  setPropTypes({
    isLoading: PropTypes.bool, // display loading placeholder
    ...Text.propTypes,
  }),
  styles,
  withPlaceholder(Typography, { width: '100%' }),
  pure,
)(Text);

export default H3;
