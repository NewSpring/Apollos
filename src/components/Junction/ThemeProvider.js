import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_THEME } from './constants';

export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.shape({
      primaryColor: PropTypes.string,
      secondaryColor: PropTypes.string,
    }),
  };

  static defaultProps = {
    children: null,
    theme: DEFAULT_THEME,
  };

  static childContextTypes = {
    theme: PropTypes.shape({
      primaryColor: PropTypes.string,
      secondaryColor: PropTypes.string,
    }),
  };

  getChildContext = () => ({
    theme: this.props.theme,
  })

  render() {
    return Children.only(this.props.children);
  }
}
