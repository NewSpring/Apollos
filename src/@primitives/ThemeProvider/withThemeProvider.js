import { mapProps, withContext, compose } from 'recompose';
import PropTypes from 'prop-types';
import { THEME_PROPS } from '@primitives/constants';
import withTheme from '@primitives/withTheme';

// HOC to help with extending theme.
// Use to over-ride theme context for child components
// Does not change or modify props of the component you're enhancing
const withThemeProvider = themeInput => compose(
  mapProps(props => ({ ownProps: props })),
  withTheme(),
  withContext({ theme: PropTypes.shape(THEME_PROPS) }, ({ theme, ownProps }) => {
    let composedTheme = Object.assign({}, theme, ownProps.theme || {});

    let themeInputAsObject = themeInput;
    if (typeof themeInput === 'function') themeInputAsObject = themeInput({ ...ownProps, theme: composedTheme });

    composedTheme = Object.assign({}, composedTheme, themeInputAsObject);

    return ({
      theme: composedTheme,
    });
  }),
  mapProps(({ ownProps }) => ownProps),
);

export default withThemeProvider;
