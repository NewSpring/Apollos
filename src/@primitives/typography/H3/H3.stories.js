import React from 'react';

import { storiesOf } from '@storybook/react-native';

import ThemeProvider from '@primitives/ThemeProvider';
import H3 from './';

storiesOf('@primitives/typography', module)
  .add('H3', () => (
    <ThemeProvider>
      <H3>{'"The Christian shoemaker does his duty not by putting little crosses on the shoes, but by making good shoes, because God is interested in good craftsmanship." â€• Martin Luther'}</H3>
    </ThemeProvider>
  ));
