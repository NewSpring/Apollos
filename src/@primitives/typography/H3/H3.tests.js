import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from '@primitives/ThemeProvider';
import H3 from './';

describe('the H3 component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <H3>Default H3 text</H3>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmon = { color: 'salmon' };
    const tree = renderer.create(
      <ThemeProvider>
        <H3 style={salmon}>Salmon text</H3>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <H3 accessible={false}>{'"What you are is God\'s gift to you, what you become is your gift to God." â€• Hans Urs von Balthasar'}</H3>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
