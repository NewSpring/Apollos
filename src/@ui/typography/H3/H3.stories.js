import React from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react-native';


import H3 from './';

storiesOf('@ui/typography/H3', module)
  .add('Default', () => (
    <H3>{'"The Christian shoemaker does his duty not by putting little crosses on the shoes, but by making good shoes, because God is interested in good craftsmanship." ― Martin Luther'}</H3>
  ))
  .add('Border Box – platform testing', () => {
    const border = { borderWidth: 1, borderStyle: 'solid' };

    return (
      <View>
        <H3 style={border}>Heading 3</H3>
        <H3 style={border}>{'"The Christian shoemaker does his duty not by putting little crosses on the shoes, but by making good shoes, because God is interested in good craftsmanship." ― Martin Luther'}</H3>
      </View>
    );
  });
