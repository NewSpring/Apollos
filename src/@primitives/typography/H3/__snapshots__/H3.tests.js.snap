// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the H3 component should accept additional props 1`] = `
<Text
  accessible={false}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#505050",
        "fontFamily": null,
        "fontSize": 32.4,
        "fontWeight": "700",
        "lineHeight": 41.22,
      },
      Object {},
    ]
  }
>
  "The Christian shoemaker does his duty not by putting little crosses on the shoes, but by making good shoes, because God is interested in good craftsmanship." â€• Martin Luther
</Text>
`;

exports[`the H3 component should accept and render passed in styles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#505050",
        "fontFamily": null,
        "fontSize": 32.4,
        "fontWeight": "700",
        "lineHeight": 41.22,
      },
      Object {
        "color": "salmon",
      },
    ]
  }
>
  Salmon text
</Text>
`;

exports[`the H3 component should render correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#505050",
        "fontFamily": null,
        "fontSize": 32.4,
        "fontWeight": "700",
        "lineHeight": 41.22,
      },
      Object {},
    ]
  }
>
  Default H3 text
</Text>
`;
